### PR TITLE
add the feature where the user would be logged out when he is idle fo…

### PR DESCRIPTION
**Pull request by beaugogh**
…r too long

### CODE REVIEW

File: src/app/components/navbar/navbar.controller.js [hunk 1]
```diff
@@ -25,4 +38,14 @@ angular.module('transmartBaseUi')
                 }
             });
 
-        }]);
+        }])
+    .config(function (IdleProvider, KeepaliveProvider) {
+        IdleProvider.idle(20*60); // 20min in seconds
```
Comment (rnugraha) [inline src/app/components/navbar/navbar.controller.js]: Can we put this in a config? Preferably in config.json

File: src/app/components/navbar/warning-dialog.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="modal-header">
+    <h3>You have been idle... </h3>
+</div>
+<div idle-countdown="countdown" ng-init="countdown=5" class="modal-body">
+    <uib-progressbar max="5" value="5" animate="false" class="progress-striped active">You'll be logged out in {{countdown}} second(s).</uib-progressbar>
```
Comment (rnugraha) [inline src/app/components/navbar/warning-dialog.html]: too long. Maintain max 120 chars

File: src/app/components/navbar/navbar.controller.js [hunk 3]
```diff
@@ -25,4 +38,14 @@ angular.module('transmartBaseUi')
                 }
             });
 
-        }]);
+        }])
+    .config(function (IdleProvider, KeepaliveProvider) {
```
Comment (rnugraha) [inline src/app/components/navbar/navbar.controller.js]: keep camelCase styling for variable/argumen naming

File: src/app/components/navbar/navbar.controller.js (hunk 1 above)
Comment (beaugogh) [reply src/app/components/navbar/navbar.controller.js]: I would prefer this way as the idle module and its parameters (e.g. the 20*60 idle time) are dedicated for the logout feature in navbar, moving the config out of its context would be confusing in future.

Comment (rnugraha) [reply src/app/components/navbar/navbar.controller.js]: It's more confusing when you need to search where to set idle time somewhere in the code. If we are using your philosophy what's the point of having config file.

Comment (beaugogh) [reply src/app/components/navbar/navbar.controller.js]: config file should be used when there is multiple use cases spread across different code segments, not whenever there is something to configure, you stash it to config file. In this case, idle is only relevant for managing logging-in/out. In future, when the programmer inspects logging and idle behavior, he can immediately find the config right beside the controller's code, not in some distant config file. But that is just my philosophy.

Comment (rnugraha) [reply src/app/components/navbar/navbar.controller.js]: You know that hardcoding is bad practice. We've seen to much in this project. 20 * 60. what is that? and then followed by a narration to explain it is not pretty. on top of that, Controller ideally should not be the place to configure something.

Better approach:

- Config file
- Constants 

`
idleProvider.idle(IDLE_IN_MINUTES*60);
`

File: src/app/components/navbar/navbar.controller.js (hunk 3 above)
Comment (beaugogh) [reply src/app/components/navbar/navbar.controller.js]: these are preserved key words

**Pull request merged**